ui/src/app/pages/data/criteria-search.tsx

```diff
@@ -126,16 +128,30 @@ export class CriteriaSearch extends React.Component<Props, State>  {
       mode = 'tree';
     }
     this.setState({backMode, hierarchyNode, mode});
-    this.subscription = currentCohortCriteriaStore.subscribe(currentCohortCriteria => {
-      if (source === 'criteria') {
+    if (source === 'conceptSetDetails') {
+      this.getConceptSet();
+    }
+    if (source === 'criteria') {
+      this.subscription = currentCohortCriteriaStore.subscribe(currentCohortCriteria => {
         this.setState({selectedCriteriaList: currentCohortCriteria});
-      }
-    });
-    this.subscription.add(currentConceptStore.subscribe(currentConcepts => {
-      if (source === 'concept') {
+      });
+    } else {
+      this.subscription = currentConceptStore.subscribe(currentConcepts => {
         this.setState({selectedCriteriaList: currentConcepts});
-      }
-    }));
+      });
+    }
+  }
+
+  async getConceptSet() {
+    const {urlParams: {ns, wsid, csid}} = this.props;
+    try {
+      const resp = await conceptSetsApi().getConceptSet(ns, wsid, csid);
+      currentConceptStore.next(resp.criteriums);
+    } catch (error) {
+      console.log(error);
+      // TODO: what do we do with resources not found?  Currently we just have an endless spinner
+      // Maybe want to think about designing an AoU not found page for better UX
+    }
   }
 
   componentWillUnmount() {
```

Comment:
It won't let me comment on the actual line, but in the `isConcept` function below at line 168, we need to check if the source is 'conceptSetDetails' also. Currently, if you're on the detail page and try to search and add new concepts, it will get added to the criteria store and we get double criteria icons in the sidebar.

<img width="1284" alt="Screen Shot 2020-10-16 at 2 01 12 PM" src="https://user-images.githubusercontent.com/40036095/96299860-4e7bc500-0fba-11eb-9339-804b0d070e7f.png">

Reply:
Done